### ISSUE TITLE
Router: link urls doesn't have query string parameters

### ISSUE THREAD

**Issue by alexpods** [angular/angular#7367]
Copy of @zanettin's issue from https://github.com/angular/router/issues/397

When I click on a routerLink

**LINK**:
```html
<a [routerLink]=" ['./RouteParamsTest', { userId : '1919', lang : 'it', token : 'jwt' }] ">Route Params Test 2</a>
```

**CONFIG**:
```js
{ path: '/route-params-test/:userId/:lang', component: RouteParamsTest, name: 'RouteParamsTest' }
```

with disabled JS, this URL gets called => http://localhost:3000/route-params-test/1919/it and the query string won't appear. With activated JS everything works as expected.

The reason of the problem is [here](https://github.com/angular/angular/blob/master/modules/angular2/src/router/instruction.ts#L162). As you can see `urlParams` are used only in `_stringifyMatrixParams()` which is called only if we have `auxInstructions` (see [here](https://github.com/angular/angular/blob/master/modules/angular2/src/router/instruction.ts#L198)). So if we don't have auxilary routes then `urlParams` will never be used. (`toLinkUrl()` is called [here](https://github.com/angular/angular/blob/master/modules/angular2/src/router/directives/router_link.ts#L65) to update `href` of the link)

@matsko @btford @IgorMinar

**Comment by btford** [comment]
@petebacondarwin was this possibly caused by your recent refactoring?

**Comment by petebacondarwin** [comment]
I doubt the refactoring is the cause, since we only just merged the refactoring into master yesterday.
It might even be that the refactoring helps with this.  @alexpods and @zanettin can you try with the latest Router bits in Angular 2 master?

**Comment by alexpods** [comment]
@btford @petebacondarwin 

Refactoring is definitely not the cause. The original @zanettin's issue was created 21 days ago. @petebacondarwin I doubt that it will work in the latest master (actually I'm sure it won't).

The cause of the error:

1. `instruction.toLinkUrl()` method is used to create `href` attributes [here](https://github.com/angular/angular/blob/master/modules/angular2/src/router/directives/router_link.ts#L65)
2. look at [`tooLinkUrl`](https://github.com/angular/angular/blob/master/modules/angular2/src/router/instruction.ts#L161) implementation:  `urlParams` are used only as a result of `this._stringifyAux()` invokation.
3. we don't have `auxInstructions` in this case, so `urlParams` won't be applied.

**Comment by petebacondarwin** [comment]
OK I accept that it is a bug.

Just for my sake, can you explain a bit about why this error does not appear when you have JS enabled? Something different going on in Universal?

**Comment by alexpods** [comment]
@petebacondarwin 
When javascript is enabled and you click on a link [`navigateByInstruction()`](https://github.com/angular/angular/blob/master/modules/angular2/src/router/directives/router_link.ts#L79)  is used. So the user will be navigated to the correct page/state. 

When javascript is disabled `href` value will be used to move user on another page (simple `<a>` link). Because `href` value doesn't have query string, result page url also won't have query string.

**Comment by petebacondarwin** [comment]
I have created a fix for this at #7431. @btford can you give it a check, in case I have missed something.

**Comment by alexpods** [comment]
Thanks, @petebacondarwin .

Just to demonstrate the issue I've created [this plunk](http://plnkr.co/edit/AeN6I5QghTQbE67rRktk?p=preview). Look at `href` attribute of a link. It doesn't have query string with `param1` and `param2`.
